### PR TITLE
Jfr service unit tests

### PR DESCRIPTION
**Pull request by XiXiaPdx**

### CODE REVIEW

File: gradle/script/java.gradle [hunk 1]
```diff
@@ -73,8 +73,8 @@ compileTestJava.options.forkOptions.executable = jdk8 + '/bin/javac'
 compileTestJava.options.compilerArgs += '-proc:none'
 
 compileTestJava {
-    sourceCompatibility = JavaVersion.VERSION_1_7
-    targetCompatibility = JavaVersion.VERSION_1_7
+    sourceCompatibility = JavaVersion.VERSION_1_8
```
Comment (XiXiaPdx) [inline gradle/script/java.gradle:76]: can. use Java 8 in unit tests

File: gradle/script/java.gradle [hunk 2]
```diff
@@ -186,7 +186,7 @@ test {
 
 dependencies {
     testImplementation("junit:junit:4.12")
-    testImplementation("org.mockito:mockito-core:2.28.2")
+    testImplementation("org.mockito:mockito-inline:3.9.0")
```
Comment (XiXiaPdx) [inline gradle/script/java.gradle:189]: inline has the new mockito feature to mock statics

File: newrelic-agent/src/main/java/com/newrelic/agent/jfr/JfrService.java [hunk 3]
```diff
@@ -49,15 +52,15 @@ protected void doStart() {
...
+                jfrController = SetupUtils.buildJfrController(daemonConfig, uploader);
 
                 ExecutorService jfrMonitorService = Executors.newSingleThreadExecutor();
                 jfrMonitorService.submit(
                         new Runnable() {
                             @Override
                             public void run() {
                                 try {
-                                    jfrController.loop();
+                                    startJfrLoop();
```
Comment (XiXiaPdx) [inline newrelic-agent/src/main/java/com/newrelic/agent/jfr/JfrService.java:63]: In tests, I wanted to verify the loop was started and no exception. By wrapping this in a local method of this class, I can spy and verify interactions.

File: newrelic-agent/src/main/java/com/newrelic/agent/jfr/JfrService.java [hunk 4]
```diff
@@ -81,10 +89,11 @@ public final boolean isEnabled() {
 
     @Override
     protected void doStop() {
-
+        jfrController.shutdown();
```
Comment (XiXiaPdx) [inline newrelic-agent/src/main/java/com/newrelic/agent/jfr/JfrService.java:92]: seemed appropriate and also handy in test to shutdown jfr

File: newrelic-agent/src/test/java/com/newrelic/agent/language/SourceLanguageServiceTest.java [hunk 5]
```diff
@@ -29,7 +29,7 @@
 import static org.junit.Assert.assertTrue;
 
 public class SourceLanguageServiceTest {
-    private static final String SUPPORTABILITY_METRIC_FOR_CLASS = format(MetricNames.SUPPORTABILITY_LOADED_CLASSES_SOURCE_VERSION, "java", "1.7");
+    private static final String SUPPORTABILITY_METRIC_FOR_CLASS = format(MetricNames.SUPPORTABILITY_LOADED_CLASSES_SOURCE_VERSION, "java", "1.8");
```
Comment (XiXiaPdx) [inline newrelic-agent/src/test/java/com/newrelic/agent/language/SourceLanguageServiceTest.java:32]: Java 8 used now for unit tests,  so this string used for assertions need to match

**Review by jasonjkeller — APPROVED** [review]
Looks good!

**Pull request merged**